### ISSUE TITLE
NPE when executing HTTP2 service

### ISSUE THREAD

**Issue by sashikamw** [ballerina-platform/ballerina-lang#7877]
**Description:**
An NPE observed when executing the HTTP2 service

**Steps to reproduce:**
1. Start the attached HTTP2 service
2. Invoke the service using a tool like jmeter

**Affected Versions:**
ballerina-runtime-0.970.0-beta13

**OS, DB, other environment details and versions:**
Ubuntu 15.10

**Observations**
Did not observe any failures in the request invocations
 Following WARN logs appeared in the bre.log
```
[2018-04-24 11:29:20,912] WARN  {io.netty.util.concurrent.AbstractEventExecutor} - A task raised an exception. Task: org.wso2.transport.http.netty.sender.http2.ClientOutboundHandler$Http2RequestWriter$$Lambda$353/1921731469@53f495fc 
java.lang.NullPointerException
	at org.wso2.transport.http.netty.sender.http2.ClientOutboundHandler$Http2RequestWriter.lambda$null$25(ClientOutboundHandler.java:144)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:886)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:745)
 
[2018-04-24 11:35:50,410] WARN  {io.netty.util.concurrent.AbstractEventExecutor} - A task raised an exception. Task: org.wso2.transport.http.netty.sender.http2.ClientOutboundHandler$Http2RequestWriter$$Lambda$353/1921731469@6268fc8 
java.lang.NullPointerException
 
```

Note - This can be reproducible even for a limited load (2 threads).

[Artifacts.zip](https://github.com/ballerina-platform/ballerina-lang/files/1941064/Artifacts.zip)

**Comment by isudana** [comment]
This should get fixed with http transport version v6.0.156 having the fix done in 
https://github.com/wso2/transport-http/pull/144.

This was appeared as the httpContent filling logic in HTTPCaronMessage has been changed with https://github.com/ballerina-platform/ballerina-lang/pull/7780

**Comment by isudana** [comment]
http transport is upgraded with https://github.com/ballerina-platform/ballerina-lang/pull/8015, hence closing the issue.
